src/state/apis/nft/nftApi.ts

```diff
@@ -58,6 +63,126 @@ export const initialState: NftState = {
   },
 }
 
+type PortfolioAndAssetsUpsertPayload = {
+  resultsData: NftItemWithCollection[]
+  portfolioAssetIds: AssetId[]
+}
+
+const upsertPortfolioAndAssets = createAsyncThunk<void, PortfolioAndAssetsUpsertPayload>(
+  'nft/upsertPortfolioAndAssets',
+  ({ resultsData, portfolioAssetIds }, { dispatch }) => {
+    const nftsById = resultsData.reduce<NftState['nfts']['byId']>((acc, item) => {
+      if (!item.collection.assetId) return acc
+
+      const nftAssetId: AssetId = `${item.collection.assetId}/${item.id}`
+      let { collection, ...nftItemWithoutId } = item
+      const nftItem: NftItem = {
+        ...nftItemWithoutId,
+        collectionId: item.collection.assetId,
+      }
+      acc[nftAssetId] = nftItem
+
+      return acc
+    }, {})
+
+    const collectionsById = resultsData.reduce<NftState['collections']['byId']>((acc, item) => {
+      if (!item.collection.assetId) return acc
+
+      const collectionAssetId = item.collection.assetId
+      if (!collectionAssetId) return acc
+
+      acc[collectionAssetId] = item.collection
+
+      return acc
+    }, {})
+
+    dispatch(
+      nft.actions.upsertCollections({
+        byId: collectionsById,
+        ids: Object.keys(collectionsById),
+      }),
+    )
+    dispatch(nft.actions.upsertNfts({ byId: nftsById, ids: Object.keys(nftsById) }))
+
+    const portfolioNfts = portfolioAssetIds.filter(isNft) // Note: You will need to define `isNft` somewhere or replace this with the actual condition
```

Comment:
can always make a `selectPortfolioNftAssetIds` that we can leverage throughout as needed